packages/dd-trace/src/llmobs/index.js

```diff
@@ -46,7 +52,20 @@ function enable (config) {
 
   // distributed tracing for llmobs
   injectCh.subscribe(handleLLMObsParentIdInjection)
-  telemetry.recordLLMObsEnabled(startTime, config)
+
+  setAgentStrategy(config, useAgentless => {
+    if (useAgentless && !(config.apiKey && config.site)) {
+      throw new Error(
+        'Cannot send LLM Observability data without a running agent or without both a Datadog API key and site.\n' +
+        'Ensure these configurations are set before running your application.'
+      )
+    }
+
+    evalWriter?.setAgentless(useAgentless)
+    spanWriter?.setAgentless(useAgentless)
```

Comment:
Any specific reason we are setting the writers to be agentless here instead of just creating new writers once with the correct agentless mode flag?

Reply:
we want to be able to enqueue data to the writers while the async operation to get the agent info is happening. it's unlikely but possible some data is enqueued between llmobs enabled/starting the operation to get the agent info and then us finally getting it back, and making new writers at that point would essentially erase the buffers they had accumulated.